### DISCUSSION TITLE
How do I get the event that start collecting upkeepocst?

### DISCUSSION THREAD

**Discussion by LunarECL** [TownyAdvanced/Towny#5195] [Q&A] [answered]
In Towny API Is there is way to know the upkeepcost charing time? so I can do something with that time.

**Comment by LlmDl** [comment, marked as answer]
Towny uses the following when `/towny time` is used: 
```java
TownyMessaging.sendMsg(Translation.of("msg_time_until_a_new_day") + TimeMgmt.formatCountdownTime(TownyTimerHandler.townyTime()));
```

There is also a PreNewDayEvent thrown before the upkeep/newday code is run, and a NewDayEvent when it is done.
